clcache.py

```diff
@@ -1165,7 +1253,11 @@ def processCompileRequest(cache, compiler, args):
 
 def processDirect(cache, outputFile, compiler, cmdLine, sourceFile):
     manifestHash = ObjectCache.getManifestHash(compiler, cmdLine, sourceFile)
-    with cache.lock:
+    postProcessing = None
+
+    try:
```

Comment:
Not sure I get this change: why couldn't you go with `with cache.lock` anymore but needed to explicitly `acquire`/`release` the lock? Something to do with scoping?